src/main/java/com/example/sixthmafiabot/repository/UserRepository.java

```diff
@@ -1,19 +1,44 @@
 package com.example.sixthmafiabot.repository;
 
+import com.example.sixthmafiabot.DTO.UserDTO;
 import com.example.sixthmafiabot.models.User;
-import org.springframework.data.repository.Repository;
-import org.springframework.scheduling.annotation.Async;
+import com.example.sixthmafiabot.repository.Abstract.BaseRepository;
+import com.example.sixthmafiabot.repository.Abstract.SpringRepositoryImplementations.SpringUserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
 import java.util.concurrent.CompletableFuture;
 
+@Repository
 
-public interface UserRepository extends Repository<User, Long> {
+@Slf4j
+public class UserRepository implements BaseRepository {
 
+    private final ModelMapper modelMapper = new ModelMapper();
 
-    @Async("asyncExecutor")
-    void save(User user);
+    @Autowired
+    SpringUserRepository userRepository;
 
-    @Async("asyncExecutor")
-    CompletableFuture<User> getUserByTelegramId(Long telegramId);
+    public CompletableFuture<User> create(UserDTO userDTO) {
+
+        User user = modelMapper.map(userDTO, User.class);
+
+        save(user);
+        return CompletableFuture.completedFuture(user);
+    }
+
+    public void save(User user) {
```

Comment:
do we really need this method? can we call save from base repo instead?

Reply:
+